tests/test_detection.py

```diff
@@ -0,0 +1,110 @@
+import cv2
+import rpy_apriltag
+from wpimath.geometry import Transform3d
+
+import math
+import pathlib
+import pytest
+
+
+def _load_grayscale_image(fname):
+    full_path = pathlib.Path(__file__).parent / fname
+    img = cv2.imread(str(full_path))
+    return cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+
+
+def test_1():
+    detector = rpy_apriltag.AprilTagDetector()
+    assert detector.addFamily("tag16h5")
+    assert detector.addFamily("tag36h11")
+
+    img = _load_grayscale_image("tag1_640_480.jpg")
+    results = detector.detect(img)
+
+    assert len(results) == 1
+    assert results[0].getFamily() == "tag36h11"
+    assert results[0].getId() == 1
+    assert results[0].getHamming() == 0
+
+    estimator = rpy_apriltag.AprilTagPoseEstimator(
+        rpy_apriltag.AprilTagPoseEstimator.Config(0.2, 500, 500, 320, 240)
+    )
+
+    est = estimator.estimateOrthogonalIteration(results[0], 50)
+    assert est.pose2 == Transform3d()
+    pose = estimator.estimate(results[0])
+    assert est.pose1 == pose
+
+
+def test_pose_rotated_x():
+    """
+    This tag is rotated such that the top is closer to the camera than the bottom. In the camera
+    frame, with +x to the right, this is a rotation about +X by 45 degrees.
+    """
+
+    detector = rpy_apriltag.AprilTagDetector()
+    assert detector.addFamily("tag16h5")
+
+    img = _load_grayscale_image("tag2_45deg_X.png")
+    results = detector.detect(img)
+
+    assert len(results) == 1
+
+    estimator = rpy_apriltag.AprilTagPoseEstimator(
+        rpy_apriltag.AprilTagPoseEstimator.Config(
+            0.2, 500, 500, img.shape[1] / 2.0, img.shape[0] / 2.0
+        )
+    )
+    est = estimator.estimateOrthogonalIteration(results[0], 50)
+    assert pytest.approx(est.pose1.rotation().x, abs=0.1) == math.radians(45)
+    assert pytest.approx(est.pose1.rotation().y, abs=0.1) == math.radians(0)
+    assert pytest.approx(est.pose1.rotation().z, abs=0.1) == math.radians(0)
```

Comment:
If you need to specify the tolerances, probably better to just use [`math.isclose`]?

[`math.isclose`]: https://docs.python.org/3/library/math.html#math.isclose

Reply:
`pytest.approx` is the pytest way to do this. I have to specify the tolerances to match what the original WPILib test has in it, as it specifies those tolerances.